### PR TITLE
Refactor DefaultHttpClient to use ByteBody for the request body

### PR DESCRIPTION
**Pull request by yawkat**
This is an intermediate PR of my refactor, in the interest of keeping the changes reviewable.

### CODE REVIEW

File: http-netty/src/main/java/io/micronaut/http/netty/NettyHttpRequestBuilder.java [hunk 1]
```diff
@@ -74,12 +75,25 @@ default HttpRequest toHttpRequest() {
...
     }
 
+    /**
+     * Directly convert this request body to a {@link ByteBody}, if possible. If the body of this
+     * request has been changed, this will return an empty value.
+     *
+     * @return The body
+     */
+    @NonNull
+    default Optional<ByteBody> byteBodyDirect() {
```
Comment (dstepanov) [inline http-netty/src/main/java/io/micronaut/http/netty/NettyHttpRequestBuilder.java:93]: For internal API I would just use nullable type

**Review by dstepanov — APPROVED** [review]

File: http-netty/src/main/java/io/micronaut/http/netty/NettyHttpRequestBuilder.java (hunk 1 above)
Comment (yawkat) [reply http-netty/src/main/java/io/micronaut/http/netty/NettyHttpRequestBuilder.java:93]: Fair but I don't want to deal with merge conflicts now :D I will do it in a future PR

Comment (yawkat) [reply http-netty/src/main/java/io/micronaut/http/netty/NettyHttpRequestBuilder.java:93]: included in #11182

**Review by graemerocher — APPROVED** [review]
Does this solve turning the body into an input stream?

**Comment by yawkat** [comment]
No, not yet. There is no API change. But it does make it easier, because now in principle we just need a conversion ByteBody->InputStream that just does `byteBody.toInputStream()`.

**Pull request merged**